vpr/src/route/bucket.cpp

```diff
@@ -119,43 +158,48 @@ void Bucket::check_scaling() {
     auto max_bucket = cost_to_int(max_cost);
 
     // If scaling is invalid or more than 100k buckets are needed, rescale.
-    if (min_bucket < 0 || max_bucket < 0 || max_bucket > 1000000) {
-        // Choose a scaling factor that accomidates 50k buckets between
-        // min_cost_ and max_cost_.
-        //
-        // If min and max are close to each other, assume 3 orders of
-        // magnitude between min and max.  The goal is to rescale less often
-        // when the larger costs haven't been seen yet.
-        //
-        // If min and max are at least 3 orders of magnitude apart, scale
-        // soley based on max cost.  The goal at this point is to keep the
-        // number of buckets between 50k and 100k.
-        //
-        // NOTE:  The precision loss of the bucket approximation grows as the
-        // maximum cost increases.  The underlying assumption of this scaling
-        // algorithm is that the maximum cost will not result in a poor
-        // scaling factor such that all precision is lost.
-        conv_factor_ = rescale_func();
-        check_conv_factor();
-
-        // Reheap after adjusting scaling.
-        if (heap_head_ != std::numeric_limits<size_t>::max()) {
-            std::vector<BucketItem*> reheap;
-            for (size_t bucket = heap_head_; bucket <= heap_tail_; ++bucket) {
-                for (BucketItem* item = heap_[bucket]; item != nullptr; item = item->next_bucket) {
-                    reheap.push_back(item);
-                }
-            }
+    if (min_bucket < 0 || max_bucket < 0 || max_bucket > max_buckets_) {
+        rescale();
+    }
+}
 
-            std::fill(heap_ + heap_head_, heap_ + heap_tail_ + 1, nullptr);
-            heap_head_ = std::numeric_limits<size_t>::max();
-            heap_tail_ = 0;
+void Bucket::rescale() {
+    // Choose a scaling factor that accomidates 50k buckets between
```

Comment:
This comment doesn't seem to reflect the new dynamic rescaling of the number of buckets (and the space between them).